internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -655,6 +655,75 @@ func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
 	}
 }
 
+func (r *ClusterDeploymentsReconciler) getEncodedCACert(ctx context.Context,
+	log logrus.FieldLogger,
+	caCertificateRef *hiveext.CaCertificateReference) (*string, error) {
+	caSecret, err := getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: caCertificateRef.Namespace, Name: caCertificateRef.Name})
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting ca certificate secret")
+	}
+	caCertBytes, hasCACert := caSecret.Data[corev1.TLSCertKey]
+	if !hasCACert {
+		return nil, fmt.Errorf("CA Secret is missing tls.crt key")
+	}
+	encodedCACert := base64.StdEncoding.EncodeToString(caCertBytes)
+	caCertificate := swag.String(encodedCACert)
+	return caCertificate, nil
+}
+
+func (r *ClusterDeploymentsReconciler) parseIgnitionEndpoint(ctx context.Context,
+	log logrus.FieldLogger,
+	kubeapiIgnitionEndpoint *hiveext.IgnitionEndpoint) (*models.IgnitionEndpoint, error) {
+
+	ignitionEndpoint := &models.IgnitionEndpoint{}
+	if kubeapiIgnitionEndpoint.Url != "" {
+		ignitionEndpoint.URL = swag.String(kubeapiIgnitionEndpoint.Url)
+	} else {
+		ignitionEndpoint.URL = swag.String("")
+	}
+	caCertificateReference := kubeapiIgnitionEndpoint.CaCertificateReference
+	if caCertificateReference != nil {
+		caCertificate, err := r.getEncodedCACert(ctx, log, caCertificateReference)
+		if err == nil {
+			ignitionEndpoint.CaCertificate = caCertificate
+		} else {
+			return nil, err
+		}
+	} else {
+		ignitionEndpoint.CaCertificate = swag.String("")
+	}
+	return ignitionEndpoint, nil
+}
+
+func (r *ClusterDeploymentsReconciler) updateIgnitionInUpdateParams(ctx context.Context,
+	log logrus.FieldLogger,
+	clusterInstall *hiveext.AgentClusterInstall,
+	cluster *common.Cluster,
+	params *models.V2ClusterUpdateParams) bool {
+	update := false
+	if clusterInstall.Spec.IgnitionEndpoint != nil {
+		ignitionEndpoint, err := r.parseIgnitionEndpoint(ctx, log, clusterInstall.Spec.IgnitionEndpoint)
+		if err == nil {
+			params.IgnitionEndpoint = ignitionEndpoint
+			update = true
+		} else {
+			log.WithError(err).Errorf("Failed to get and parse ignition ca certificate %s/%s", clusterInstall.Namespace, clusterInstall.Name)
```

Comment:
We cannot ignore errors, please return the error and handle it.
It should be reflected eventually in the conditions.

Reply:
Fixed